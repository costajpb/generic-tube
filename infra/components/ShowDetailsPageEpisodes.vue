<script setup lang="ts">
    import type Episode from '@/domain/episode/entity';

    defineProps<{
        episodes: Episode[]
    }>()
</script>

<template>
    <h2>Episodes</h2>
    <ol>
        <li v-for="episode in episodes" :key="episode.id">
            <article>
                <header>
                    <h3>{{ episode.title }}</h3>
                    <p>{{ episode.duration }}m</p>
                </header>
                <img :src="episode.coverImage" :alt="episode.title" />
                <div class="summary" v-html="episode.summary" />
            </article>
        </li>
    </ol>
</template>

<style scoped lang="postcss">
    * {
        all: unset;
    }

    a {
        cursor: pointer;
    }

    h2 {
        padding-inline: var(--layout-margin-inline);
        display: block;
        font-size: var(--font-size-5);
        line-height: 2;
        border-bottom: var(--border-size-2) var(--branding-color-primary-100) solid;
    }

    ol {
        display: grid;
        /* FIXME: 300px */
        grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
        gap: var(--size-4);
        max-height: 60vh;
        overflow: auto;
        background: var(--gray-0);
        /* box-sizing: content-box; */
        /* padding: var(--layout-margin-inline); */
        /* margin-left: calc(var(--layout-margin-inline) * -1); */
        box-shadow: var(--inner-shadow-2);
        padding: var(--layout-margin-inline);

        @media (min-width: 700px) {
            max-height: none;
        }
    }

    h3 {
        font-size: var(--font-size-1)
    }

    article {
        display: grid;
        grid-template-areas:
            "e f"
            "g g"
        ;
        gap: var(--size-2);
        grid-template-columns: minmax(10ch, 20vw) auto;
        box-shadow: var(--shadow-4);
        padding: var(--size-2) var(--size-4);
        /* FIXME */
        background-color: white;
        border-radius: var(--radius-2);
        transition: transform ease 300ms;

        &:hover {
            transform: scale(102%, 102%);
        }
    }

    img {
        grid-area: e;
        width: 100%;
        height: 90%;
        object-fit: cover;
    }

    header {
        grid-area: f;
    }

    h3 {
        display: block;
        font-weight: var(--font-weight-6);
        font-size: var(--font-size-2);
        color: var(--heading-color);
    }

    .summary {
        grid-area: g;
        -webkit-line-clamp: 2;
        display: -webkit-box;
        -webkit-box-orient: vertical;
        overflow: hidden;
    }
</style>