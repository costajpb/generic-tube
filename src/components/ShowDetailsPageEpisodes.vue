<script setup lang="ts">
    import type Episode from '@/domain/episode/entity';

    defineProps<{
        episodes: Episode[]
    }>()
</script>

<template>
    <ol>
        <li v-for="episode in episodes" :key="episode.id">
            {{ episode.title }}
        </li>
    </ol>
</template>