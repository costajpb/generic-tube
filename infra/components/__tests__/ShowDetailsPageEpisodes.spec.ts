import { mount } from "@vue/test-utils"
import ShowDetailsPageEpisodes from "../ShowDetailsPageEpisodes.vue"
import clickOutside from "@/infra/directives/click-outside"

describe('infra/components/show-details-page-episodes', () => {
    it('should render', () => {
        expect(mount(ShowDetailsPageEpisodes, {
            props: {
                episodes: [{
                    title: 'Episode title',
                    coverImage: 'image',
                    duration: 56,
                    summary: 'Episode summary',
                    id: 1
                }]
            },
            global: {
                directives: {
                    'click-outside': clickOutside
                }
            }
        }).element).toMatchSnapshot()
    })
})