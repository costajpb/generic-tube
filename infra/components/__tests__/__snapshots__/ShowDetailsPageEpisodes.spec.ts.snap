// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`infra/components/show-details-page-episodes > should render 1`] = `
<div
  data-v-app=""
>
  
  <h2
    class="_title_37b976"
  >
    Episodes
  </h2>
  <ol
    class="_episodes_37b976"
  >
    
    <li>
      <article
        class="_episode_37b976"
      >
        <header
          class="_episode-header_37b976"
        >
          <h3
            class="_episode-title_37b976"
          >
            Episode title
          </h3>
          <p>
            56m
          </p>
        </header>
        <img
          alt="Episode title"
          class="_cover_37b976"
          src="image"
        />
        <div
          class="_summary_37b976"
        >
          Episode summary
        </div>
      </article>
    </li>
    
  </ol>
  
</div>
`;
